### ISSUE TITLE
Frequent AttributeError: 'NoneType' had no attribute 'decode_32bit_float' 

### ISSUE THREAD

**Issue by marienboonman** [nmakel/sdm_modbus#31]
First of all, thanks for making this package.

Using it (trying to set it up for myself) I encounter the following: 

~Half of the times I try to read the voltage (or any variable) I get the following error:
```
  File "<stdin>", line 1, in <module>
  File "/home/p1mon/.local/lib/python3.9/site-packages/sdm_modbus/meter.py", line 303, in read
    return self._read(self.registers[key])
  File "/home/p1mon/.local/lib/python3.9/site-packages/sdm_modbus/meter.py", line 216, in _read
    return self._decode_value(self._read_input_registers(address, length), length, dtype, vtype)
  File "/home/p1mon/.local/lib/python3.9/site-packages/sdm_modbus/meter.py", line 199, in _decode_value
    return vtype(data.decode_32bit_float())
AttributeError: 'NoneType' object has no attribute 'decode_32bit_float'
```

I couldn't really see what I was doing wring so I just tried to do 100 readings. As you can see below, some readings come through, some don't. There's no real pattern here, I suppose. Am I missing something?

````
while True:
...     n = n+1
...     try:
...         reading = meter.read('power_active')
...         print(reading)
...     except:
...         print('reading failed')
...     time.sleep(0.5)
...     if n >100:
...         break
...

reading failed
reading failed
271.29998779296875
reading failed
271.20001220703125
reading failed
reading failed
269.8999938964844
269.79998779296875
reading failed
269.79998779296875
reading failed
269.79998779296875
reading failed
reading failed
reading failed
reading failed
272.79998779296875
272.29998779296875
reading failed
271.6000061035156
reading failed
271.29998779296875
reading failed
272.20001220703125
reading failed
273.0
273.1000061035156
reading failed
273.1000061035156
273.0
reading failed
272.70001220703125
reading failed
272.8999938964844
272.79998779296875
273.1000061035156
reading failed
273.3999938964844
reading failed
272.8999938964844
reading failed
271.70001220703125
reading failed
272.0
reading failed
270.8999938964844
reading failed
272.29998779296875
reading failed
271.1000061035156
reading failed
271.29998779296875
reading failed
272.20001220703125
reading failed
272.6000061035156
reading failed
reading failed
271.6000061035156
271.3999938964844
reading failed
270.8999938964844
reading failed
270.20001220703125
reading failed
270.1000061035156
reading failed
269.8999938964844
reading failed
270.79998779296875
270.6000061035156
reading failed
268.70001220703125
reading failed
268.8999938964844
reading failed
268.6000061035156
reading failed
270.29998779296875
reading failed
270.5
reading failed
270.5
reading failed
271.5
reading failed
270.29998779296875
reading failed
271.5
reading failed
271.29998779296875
reading failed
270.8999938964844
reading failed
271.20001220703125
reading failed
reading failed
271.0
269.70001220703125
reading failed
````

**Comment by marienboonman** [comment]
Also ran it with a 10 second pause. More-or-less the same issue:

```
 while True:
...     try:
...         print('{} {}'.format(meter.read('voltage'), datetime.now()))
...     except:
...         print('{} {}'.format('failed at', datetime.now()))
...     time.sleep(10)
...     n = n+1
...     if n > 100:
...         break
...
failed at 2023-10-11 14:51:52.118581
235.0 2023-10-11 14:52:02.337947
234.89999389648438 2023-10-11 14:52:13.660823
234.6999969482422 2023-10-11 14:52:24.977200
234.8000030517578 2023-10-11 14:52:35.087965
234.8000030517578 2023-10-11 14:52:46.411494
failed at 2023-10-11 14:52:58.734684
235.1999969482422 2023-10-11 14:53:08.958714
235.0 2023-10-11 14:53:20.281894
failed at 2023-10-11 14:53:32.604996
235.39999389648438 2023-10-11 14:53:42.825004
failed at 2023-10-11 14:53:55.148285
failed at 2023-10-11 14:54:06.465692
235.39999389648438 2023-10-11 14:54:16.580510
235.0 2023-10-11 14:54:27.897912
failed at 2023-10-11 14:54:40.217592
235.10000610351562 2023-10-11 14:54:50.437967
235.10000610351562 2023-10-11 14:55:01.757666
234.5 2023-10-11 14:55:13.081676
234.10000610351562 2023-10-11 14:55:23.195235
failed at 2023-10-11 14:55:35.568491
failed at 2023-10-11 14:55:46.886119
failed at 2023-10-11 14:55:59.213822
failed at 2023-10-11 14:56:10.536159
failed at 2023-10-11 14:56:22.863466
234.10000610351562 2023-10-11 14:56:33.082959
234.39999389648438 2023-10-11 14:56:43.188386
234.6999969482422 2023-10-11 14:56:54.506738
234.0 2023-10-11 14:57:04.620324
234.39999389648438 2023-10-11 14:57:15.937465
234.39999389648438 2023-10-11 14:57:27.257489
234.60000610351562 2023-10-11 14:57:37.371156
234.0 2023-10-11 14:57:48.694397
failed at 2023-10-11 14:58:01.016159
234.10000610351562 2023-10-11 14:58:11.235594
233.89999389648438 2023-10-11 14:58:21.349214
233.6999969482422 2023-10-11 14:58:32.666590
233.10000610351562 2023-10-11 14:58:43.989622
233.39999389648438 2023-10-11 14:58:54.107256
233.3000030517578 2023-10-11 14:59:05.429957
failed at 2023-10-11 14:59:17.753301
failed at 2023-10-11 14:59:29.978882
failed at 2023-10-11 14:59:42.301442
failed at 2023-10-11 14:59:53.624208
234.89999389648438 2023-10-11 15:00:03.741882
failed at 2023-10-11 15:00:16.066219
failed at 2023-10-11 15:00:27.386207
failed at 2023-10-11 15:00:40.617209
failed at 2023-10-11 15:00:51.939186
failed at 2023-10-11 15:01:04.262944
failed at 2023-10-11 15:01:15.586888
233.8000030517578 2023-10-11 15:01:25.817101
failed at 2023-10-11 15:01:38.179180
failed at 2023-10-11 15:01:50.405195
failed at 2023-10-11 15:02:02.732647
failed at 2023-10-11 15:02:14.054772
234.10000610351562 2023-10-11 15:02:24.171400
233.89999389648438 2023-10-11 15:02:35.494604
failed at 2023-10-11 15:02:47.818430
232.89999389648438 2023-10-11 15:02:58.037781
233.10000610351562 2023-10-11 15:03:08.151430
233.1999969482422 2023-10-11 15:03:19.474430
234.0 2023-10-11 15:03:29.588014
234.0 2023-10-11 15:03:40.911270
failed at 2023-10-11 15:03:53.234325
234.0 2023-10-11 15:04:03.453767
failed at 2023-10-11 15:04:15.827361
failed at 2023-10-11 15:04:27.149455
234.39999389648438 2023-10-11 15:04:38.470741
234.1999969482422 2023-10-11 15:04:49.793746
failed at 2023-10-11 15:05:02.116930
233.8000030517578 2023-10-11 15:05:12.336319
233.8000030517578 2023-10-11 15:05:23.659305
233.6999969482422 2023-10-11 15:05:34.982456
234.10000610351562 2023-10-11 15:05:45.096061
233.89999389648438 2023-10-11 15:05:56.418781
failed at 2023-10-11 15:06:08.743050
failed at 2023-10-11 15:06:20.961694
failed at 2023-10-11 15:06:33.281184
failed at 2023-10-11 15:06:44.603302
234.3000030517578 2023-10-11 15:06:54.721006
failed at 2023-10-11 15:07:07.037773
failed at 2023-10-11 15:07:18.355952
failed at 2023-10-11 15:07:30.733048
failed at 2023-10-11 15:07:42.958970
failed at 2023-10-11 15:07:55.281838
234.3000030517578 2023-10-11 15:08:05.501764
234.1999969482422 2023-10-11 15:08:16.816805
failed at 2023-10-11 15:08:29.140203
234.10000610351562 2023-10-11 15:08:39.359638
234.3000030517578 2023-10-11 15:08:50.682955
failed at 2023-10-11 15:09:03.006356
234.5 2023-10-11 15:09:13.225700
234.5 2023-10-11 15:09:23.339283
234.3000030517578 2023-10-11 15:09:34.656557
failed at 2023-10-11 15:09:46.976901
failed at 2023-10-11 15:09:58.296817
failed at 2023-10-11 15:10:11.530128
failed at 2023-10-11 15:10:23.748982
failed at 2023-10-11 15:10:36.070725
failed at 2023-10-11 15:10:47.392831
```

**Comment by marienboonman** [comment]
Thank you for coming back to this.

I did run it with a sleep as well. I am connecting using a a RS485 to usb converter. However, I think I identified the issue. I was running a pi with a this image: https://www.ztatz.nl/p1-monitor-roadmap/

and apparently I grossly underestimated the combined load of reading the modbus meter and the built-in p1 monitor (in the image). 

I installed it on a 'fresh' pi and the problems went away! Sorry for not coming back with this closing the issue and thanks again for your reply!
